tests/integration/identity.go

```diff
@@ -21,31 +21,59 @@ import (
 	acpIdentity "github.com/sourcenetwork/defradb/acp/identity"
 )
 
-// identityRef is a type that refers to a specific identity of a certain type.
-type identityRef struct {
-	isClient bool
-	index    int
+type identityType int
+
+const (
+	noType identityType = iota
+	clientType
+	nodeType
+)
+
+// identity is a type that refers to a specific identity of a certain type.
+type identity struct {
+	index int
+	iType identityType
 }
 
 // NoIdentity returns an reference to an identity that represents no identity.
-func NoIdentity() immutable.Option[identityRef] {
-	return immutable.None[identityRef]()
+func NodeIdentity(index int) identity {
+	return identity{index, nodeType}
 }
 
 // ClientIdentity returns a reference to a user identity with a given index.
-func ClientIdentity(index int) immutable.Option[identityRef] {
-	return immutable.Some(identityRef{
-		isClient: true,
-		index:    index,
-	})
+func ClientIdentity(index int) identity {
+	return identity{index, clientType}
 }
 
 // NodeIdentity returns a reference to a node identity with a given index.
-func NodeIdentity(index int) immutable.Option[identityRef] {
-	return immutable.Some(identityRef{
-		isClient: false,
-		index:    index,
-	})
+func NoIdentity() identity {
+	return identity{0, noType}
+}
+
+func (i identity) get(s *state) acpIdentity.Identity {
+	var identities map[int]*identityHolder
+	switch i.iType {
+	case clientType:
+		identities = s.clientIdentities
+	case nodeType:
+		identities = s.nodeIdentities
+	default:
+		return acpIdentity.Identity{}
+	}
+	return getIdentityHolder(s, i.index, identities).Identity
+}
```

Comment:
question: Did you consider an identity interface that has a `Get` method perhaps? Maybe over abstraction perhaps but would remove the need for knowing the `iType` and the respective implementation would return/get their specific field.

Reply:
I believe adding an interface is overkill for this purpose. And the type will be known anyway, it will just take different shape.